apps/src/lab2/projects/ProjectManager.ts

```diff
@@ -320,6 +323,34 @@ export default class ProjectManager {
     this.lastSource = JSON.stringify(lastSource);
   }
 
+  getShouldCaptureThumbnail() {
+    const {isShareView} = getStore().getState().lab;
+    return this.channelId && this.lastChannel?.isOwner && !isShareView;
+  }
+
+  setThumbnail(pngBlob: Blob) {
+    this.thumbnailPngBlob = pngBlob;
+    this.thumbnailUrl = `/v3/files/${this.channelId}/.metadata/thumbnail.png`;
+  }
+
+  /**
+   * Uploads a thumbnail image to the thumbnail path via the files API.
+   */
+  saveThumbnail(): Promise<Response | void> {
+    if (this.thumbnailUrl && this.thumbnailPngBlob) {
+      return HttpClient.put(
+        this.thumbnailUrl,
```

Comment:
I'm debating whether we need to make a `filesStore`/`filesApi` similar to the channels and sources ones we have now. The reason to do it would be consistency, and because it extracts out the logic for saving so we can modify it later without changing the ProjectManager. We can also do something like we did in the early days of Music Lab, and sub in a different saving mechanism depending on the lab. 

The con is it's more boilerplate code for a simple http request. What do you think?

Reply:
I think that makes sense and will log for a follow-up.